src/keyring/KeyringController.ts

```diff
@@ -740,6 +751,114 @@ export class KeyringController extends BaseController<
     await this.#keyring.persistAllKeyrings();
     await this.fullUpdate();
   }
+
+  // Ledger Related methods
+
+  private async addLedgerKeyring(): Promise<LedgerKeyring> {
+    return await this.#keyring.addNewKeyring(KeyringTypes.ledger);
+  }
+
+  /**
+   * Retrieve the existing LedgerKeyring or create a new one.
+   *
+   * @returns The stored Ledger Keyring
+   */
+  async getLedgerKeyring(): Promise<LedgerKeyring> {
+    const keyring = this.#keyring.getKeyringsByType(KeyringTypes.ledger)[0];
+
+    if (keyring) {
+      return keyring;
+    }
+
+    return await this.addLedgerKeyring();
+  }
+
+  /**
+   * Connects to the ledger device by requesting some metadata from it.
+   *
+   * @param transport - The transport to use to connect to the device
+   * @param deviceId - The device ID to connect to
+   * @returns The name of the currently open application on the device
+   */
+  async connectLedgerHardware(
+    transport: Transport,
+    deviceId: string,
+  ): Promise<string> {
+    const keyring = await this.getLedgerKeyring();
+    keyring.setTransport(transport, deviceId);
+
+    const { appName } = await keyring.getAppAndVersion();
+
+    return appName;
+  }
+
+  /**
+   * Retrieve the first account from the Ledger device.
+   *
+   * @returns The default (first) account on the device
+   */
+  async unlockDefaultLedgerAccount(): Promise<{
```

Comment:
This method isn't being tested directly, should it? There's a bit of logic here so it might be good to confirm that this works as designed.

Reply:
[8455d26](https://github.com/MetaMask/controllers/pull/862/commits/8455d26052bb472c86b8d27edc956489f7841c61)

Reply:
Hmm, only the return value seems to be tested here. Any way we can test changes this makes to state as well as the `if` statements on line 811 and 812?